platforms/jvm/testing-jvm/src/main/java/org/gradle/api/tasks/testing/testng/TestNGOptions.java

```diff
@@ -54,6 +54,7 @@ public abstract class TestNGOptions extends TestFrameworkOptions {
     public static final String DEFAULT_CONFIG_FAILURE_POLICY = TestNGTestClassProcessor.DEFAULT_CONFIG_FAILURE_POLICY;
     private static final String DEFAULT_PARALLEL_MODE = null;
     private static final int DEFAULT_THREAD_COUNT = -1;
+    private static final int DEFAULT_SUITE_THREAD_POOL_SIZE_DEFAULT = 1;
```

Comment:
❗ I was trying to verify the TestNg default in [the docs](https://testng.org/parameters.html), which seem to say it is actually 10: "Each of the parallel data providers running from an XML file runs with a thread pool which has a size of 10 by default."

I'm not a TestNG user, but is this correct, and, if so, shouldn't Gradle also use 10?  If not, can you explain or point me to a reference for this?

Reply:
Let me clarify. 

Data providers in TestNG are methods that [supply data for tests](https://testng.org/#:~:text=Marks%20a%20method,of%20this%20annotation). When we increase its threads, we increase the number of parallel executions of the test method with different data. This is controlled by the variable [dataProviderThreadCount](https://testng.org/#:~:text=%2D-,dataproviderthreadcount,-The%20default%20number), which defaults to 10. In other words, it's almost like parallelizing tests by methods.

In my changes, I propose allowing the user to change [suiteThreadPoolSize](https://testng.org/#:~:text=for%20running%20tests.-,%2Dsuitethreadpoolsize,-An%20integer%20value ), which enables the desired number of test suites to run simultaneously. This variable defaults to 1 in TestNG.

Reply:
THanks for the explanation.